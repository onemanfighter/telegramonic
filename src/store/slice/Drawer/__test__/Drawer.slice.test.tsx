import { AlertComponentState, DrawerState, appStore } from '@store';
import { act, renderHook } from '@testing-library/react';

jest.useFakeTimers();

describe('Drawer slice', () => {
  const onDrawerCloseMock = jest.fn();
  const drawerData: DrawerState['drawerData'] = {
    title: 'title',
    isOpen: true,
    children: <div>children </div>,
    footer: <div>footer </div>,
    onModalClose: onDrawerCloseMock,
  };

  it('should return drawer selector state and actions', () => {
    const drawer = renderHook(() => appStore()).result.current;

    expect(drawer).toMatchSnapshot();
  });

  it('should return drawer state with initial value', () => {
    const { result } = renderHook(() => appStore());

    expect(result.current.Drawer.drawerData).toMatchSnapshot();
  });

  it('should return drawer state on drawer data set using openDrawer', () => {
    const { result } = renderHook(() => appStore());

    act(() => {
      result.current.Drawer.openDrawer(drawerData);
    });

    expect(result.current.Drawer.drawerData).toMatchSnapshot();
  });

  it('should reset and close drawer state on drawer data set using closeDrawer', () => {
    const { result } = renderHook(() => appStore());

    act(() => {
      result.current.Drawer.openDrawer(drawerData);
    });
    expect(result.current.Drawer.drawerData.isOpen).toBeTruthy();

    act(() => {
      result.current.Drawer.closeDrawer();
      jest.runAllTimers();
    });

    const { isOpen, title } = result.current.Drawer.drawerData;
    expect(title).toEqual('');
    expect(isOpen).toBeFalsy();
  });
});
