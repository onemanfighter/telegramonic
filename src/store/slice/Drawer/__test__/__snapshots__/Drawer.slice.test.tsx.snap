// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer slice should return drawer selector state and actions 1`] = `
{
  "Alert": {
    "alert": {
      "description": "",
      "showIcon": true,
      "status": undefined,
      "title": "",
      "variant": "subtle",
    },
    "alertVisibleState": false,
    "setAlertStateData": [Function],
    "setAlertVisible": [Function],
  },
  "Drawer": {
    "closeDrawer": [Function],
    "drawerData": {
      "children": null,
      "footer": null,
      "isOpen": false,
      "onModalClose": [Function],
      "title": "",
    },
    "openDrawer": [Function],
    "openState": "CLOSE",
  },
  "Modal": {
    "closeModal": [Function],
    "modalData": {
      "message": "",
      "onModalClose": [Function],
      "onPrimaryButtonClick": [Function],
      "primaryButtonLabel": "",
      "title": "",
    },
    "openModal": [Function],
    "openState": "CLOSE",
  },
}
`;

exports[`Drawer slice should return drawer state on drawer data set using openDrawer 1`] = `
{
  "children": <div>
    children 
  </div>,
  "footer": <div>
    footer 
  </div>,
  "isOpen": true,
  "onModalClose": [MockFunction],
  "title": "title",
}
`;

exports[`Drawer slice should return drawer state with initial value 1`] = `
{
  "children": null,
  "footer": null,
  "isOpen": false,
  "onModalClose": [Function],
  "title": "",
}
`;
